src/pytorch_lightning/utilities/meta.py

```diff
@@ -42,7 +42,7 @@
     ####################################################################
 
     @contextmanager
-    def enable_python_mode(cls) -> Iterator[None]:
+    def enable_python_mode(cls: Type[_MetaContext]) -> Iterator[None]:
```

Comment:
Hi, this seems to not work as `_MetaContext` is not defined yet. Because of that, everything is failing with `NameError`. You can either forward reference the annotation or do `from __future__ import __annotations__`

Reply:
That's an error I didn't expect! 
I resolved it in  https://github.com/Lightning-AI/lightning/pull/13763/commits/b93b627b765871209e98270809b5a371b724a628